changelog.d/11667.bugfix

```diff
@@ -0,0 +1 @@
+Fix `/_matrix/client/v1/room/{roomId}/hierarchy` endpoint returning incorrect fields.
```

Comment:
The code has existed since Synapse v1.31.0, but the stable endpoint only from 1.49.0?

```suggestion
Fix `/_matrix/client/v1/room/{roomId}/hierarchy` endpoint returning incorrect fields which have been present since Synapse 1.49.0.
```

Reply:
I'm not really sure adding a version here helps, but we've generally been trying to do that.

Reply:
I believe you!